src/store/Transactions.ts

```diff
@@ -15,93 +19,248 @@ const types = flow(
     'RESET_ACTIVE_TRANSACTIONS',
     'ADD_ACTIVE_TRANSACTION',
     'REMOVE_ACTIVE_TRANSACTION',
-    'UPDATE_ACTIVE_TRANSACTIONS',
+    'SET_TRANSACTION_DETAILS_ID',
+    'GET_ACTIVITY',
+    'SET_EXTERNAL_ACTIVITY',
+    'SET_EXTERNAL_ACTIVITY_TOTAL',
+    'SET_EXTERNAL_ACTIVITY_PAGINATION',
   ]),
   map((x) => [x, x]),
   fromPairs
 )([]);
 
-const isValidTransaction = (tx: History) => tx && tx.id !== undefined;
-
 type TransactionsState = {
-  activeTransactions: Array<History>;
+  activity: AccountHistory<HistoryItem>;
+  externalActivity: AccountHistory<HistoryItem>;
```

Comment:
We decided to work with naming a little bit more and rename all places with `activity` to `history`.